specification/web/resource-manager/Microsoft.Web/2016-03-01/Diagnostics.json

```diff
@@ -0,0 +1,1532 @@
+            }
+          },
+          "nonCorrelatedDetectors": {
+            "description": "Data by each detector for detectors that did not corelate",
+            "type": "array",
+            "items": {
+              "$ref": "#/definitions/DetectorDefinition"
+            }
+          }
+        }
+      },
+      "DiagnosticAnalysisCollection": {
+        "description": "Collection of Diagnostic Analyses",
+        "required": [
+          "value"
+        ],
+        "type": "object",
+        "properties": {
+          "value": {
+            "description": "Collection of resources.",
+            "type": "array",
+            "items": {
+              "$ref": "#/definitions/AnalysisDefinition"
+            }
+          },
+          "nextLink": {
+            "description": "Link to next page of resources.",
+            "type": "string"
+          }
+        }
+      },
+      "DiagnosticCategory": {
+        "description": "Class representing detector definition",
+        "type": "object",
+        "allOf": [
+          {
+            "$ref": "./CommonDefinitions.json#/definitions/ProxyOnlyResource"
+          }
+        ],
+        "properties": {
+          "properties": {
+            "description": "DiagnosticCategory resource specific properties",
+            "properties": {
+              "diagnosticCategoryName": {
+                "description": "Name of the diagnostic category",
+                "type": "string",
+                "readOnly": true
+              },
+              "description": {
+                "description": "Description of the diagnostic category",
+                "type": "string",
+                "readOnly": true
+              }
+            },
+            "x-ms-client-flatten": true
+          }
+        }
+      },
+      "DiagnosticCategoryCollection": {
+        "description": "Collection of Diagnostic Categories",
+        "required": [
+          "value"
+        ],
+        "type": "object",
+        "properties": {
+          "value": {
+            "description": "Collection of resources.",
+            "type": "array",
+            "items": {
+              "$ref": "#/definitions/DiagnosticCategory"
+            }
+          },
+          "nextLink": {
+            "description": "Link to next page of resources.",
+            "type": "string"
+          }
+        }
+      },
+      "DiagnosticDetectorCollection": {
+        "description": "Collection of Diagnostic Detectors",
+        "required": [
+          "value"
+        ],
+        "type": "object",
+        "properties": {
+          "value": {
+            "description": "Collection of resources.",
+            "type": "array",
+            "items": {
+              "$ref": "#/definitions/DetectorDefinition"
+            }
+          },
+          "nextLink": {
+            "description": "Link to next page of resources.",
+            "type": "string"
+          }
+        }
+      },
+      "DiagnosticDetectorResponse": {
```

Comment:
id, name type properties missing. This should also inherit from ProxyOnlyResource

Reply:
And all other properties should go inside the properties bag.